test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpExporterOptionsTests.cs

```diff
@@ -37,100 +80,149 @@ public void OtlpExporterOptions_DefaultsForHttpProtobuf()
-        Environment.SetEnvironmentVariable(OtlpExporterSpecEnvVarKeyDefinitions.DefaultHeadersEnvVarName, "A=2,B=3");
-        Environment.SetEnvironmentVariable(OtlpExporterSpecEnvVarKeyDefinitions.DefaultTimeoutEnvVarName, "2000");
-        Environment.SetEnvironmentVariable(OtlpExporterSpecEnvVarKeyDefinitions.DefaultProtocolEnvVarName, "grpc");
-
-        var options = new OtlpExporterOptions
+        var values = new Dictionary<string, string>()
         {
-            Endpoint = new Uri("http://localhost:200"),
-            Headers = "C=3",
-            TimeoutMilliseconds = 40000,
-            Protocol = OtlpExportProtocol.HttpProtobuf,
+            ["Endpoint"] = "http://test:8888",
+            ["Timeout"] = "2000",
+            ["Protocol"] = "grpc",
+            ["Headers"] = "A=2,B=3",
         };
 
+        var configuration = new ConfigurationBuilder()
+            .AddInMemoryCollection(values)
+            .Build();
+
+        var options = new OtlpExporterOptions();
+
+        options.ApplyConfigurationUsingSpecificationEnvVars(
+            configuration,
+            "Endpoint",
+            appendSignalPathToEndpoint: true,
+            "Protocol",
+            "Headers",
+            "Timeout");
+
+        options.Endpoint = new Uri("http://localhost:200");
+        options.Headers = "C=3";
+        options.TimeoutMilliseconds = 40000;
+        options.Protocol = OtlpExportProtocol.HttpProtobuf;
+
         Assert.Equal(new Uri("http://localhost:200"), options.Endpoint);
         Assert.Equal("C=3", options.Headers);
         Assert.Equal(40000, options.TimeoutMilliseconds);
         Assert.Equal(OtlpExportProtocol.HttpProtobuf, options.Protocol);
+        Assert.False(options.AppendSignalPathToEndpoint);
     }
 
     [Fact]
-    public void OtlpExporterOptions_ProtocolSetterDoesNotOverrideCustomEndpointFromEnvVariables()
+    public void OtlpExporterOptions_EndpointGetterUsesProtocolWhenNull()
     {
-        Environment.SetEnvironmentVariable(OtlpExporterSpecEnvVarKeyDefinitions.DefaultEndpointEnvVarName, "http://test:8888");
+        var options = new OtlpExporterOptions();
 
-        var options = new OtlpExporterOptions { Protocol = OtlpExportProtocol.Grpc };
+        Assert.Equal(new Uri(OtlpExporterOptions.DefaultGrpcEndpoint), options.Endpoint);
+        Assert.Equal(OtlpExporterOptions.DefaultOtlpExportProtocol, options.Protocol);
 
-        Assert.Equal(new Uri("http://test:8888"), options.Endpoint);
-        Assert.Equal(OtlpExportProtocol.Grpc, options.Protocol);
+        options.Protocol = OtlpExportProtocol.HttpProtobuf;
+
+        Assert.Equal(new Uri(OtlpExporterOptions.DefaultHttpEndpoint), options.Endpoint);
     }
 
     [Fact]
-    public void OtlpExporterOptions_ProtocolSetterDoesNotOverrideCustomEndpointFromSetter()
+    public void OtlpExporterOptions_EndpointGetterIgnoredProtocolWhenNotNull()
```

Comment:
Since we are changing the test to verify that endpoint getter ignores protocol when it's not null, then maybe we should update the protocol again from `Grpc` to `HttpProtobuf` and verify that the endpoint is still `"http://test:8888"`.